### PR TITLE
[release-4.8] Bug 1998938: Use client-go's leader election implementation

### PR DESCRIPTION
**Pull request by timflannagan**
**Description of the change:**

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -112,120 +137,117 @@ func main() {
...
-	// migrate away from Marketplace API
-	clientGo, err := client.New(cfg, client.Options{Scheme: mgr.GetScheme()})
-	if err != nil && !k8sErrors.IsNotFound(err) {
-		logrus.Fatal(err, "Failed to instantiate the client for migrator")
-	}
-	migrator := migrator.New(clientGo)
-	err = migrator.Migrate()
-	if err != nil {
-		logrus.Error(err, "[migration] Error while migrating Marketplace away from OperatorSource API")
-	}
```
Comment (timflannagan) [inline cmd/manager/main.go:186]: Quick note to reviewers: the master/release-4.9 PR automatic cherrypick failed when attempting to apply these commit(s) to the existing cmd/manager/main.go implementation. As a result, I added the commit for removing the migration logic that was added during the 4.9 release timeframe: https://github.com/operator-framework/operator-marketplace/pull/421/commits/bcb928902743bf8f7942d721049be34372d6a659. Should be a safe change to sneak in given that OperatorSource APIs were deprecated in 4.5 and removed in 4.6+.

File: pkg/signals/signals.go [hunk 2]
```diff
@@ -28,7 +30,9 @@ func Context() context.Context {
 
 			select {
 			case <-signalCtx.Done():
+				logrus.Info("received the done signal")
```
Comment (timflannagan) [inline pkg/signals/signals.go:33]: Hmm, these look like debug artifacts that were added in the initial, merged PR that made it's way through code review. I'm not entirely sure I see the value in logging the current state of signal handling, at least at the info-level log level, but not the end of the world. Happy to modify that cherrypick(-ed) commit and remove this additional logging if anyone has a strong opinion.

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

**Comment by mfojtik** [comment]
[patch-manager] :rocket: Approved for z-stream by score: 0.50

adding cherry-pick approved

**Pull request merged**